### ISSUE TITLE
[Proof] Claim & proof lifecycle testing

### ISSUE THREAD

**Issue by bryanchriswhite** [pokt-network/poktroll#403]
## Objective

1. Add ~~unit~~ _integration_ coverage for proof module's `#SubmitProof()` method (analogous to `msg_server_create_claim_test.go`).
2. Add ~~integration~~ _CLI_ coverage to claim/proof lifecycle ~~sufficient for testing regression of minimum-viable-testnet~~ (**depends on #404**).

## Origin Document

(See: `x/proof/module/tx_create_claim_test.go`)
![image](https://github.com/pokt-network/poktroll/assets/600733/fff89482-8504-4105-a0e0-80a5480b35f9)

(See: `x/proof/module/tx_submit_proof_test.go`)
![image](https://github.com/pokt-network/poktroll/assets/600733/b58c2b5c-e365-4522-8c7b-c170221c3e00)

## Goals

- Ensure comprehensive unit and integration coverage exists for the claim/proof lifecycle sufficient to detect regression of existing validations.

## Deliverables

- [x] A PR which adds integration coverage of claim creation.
  - #412
- [x] A PR which adds integration coverage of proof submission.
  - #417
- [ ] A PR which adds test coverage of claim creation via the CLI
- [ ] A PR which adds test coverage of proof submission via the CLI

## Non-goals / Non-deliverables

- Refactoring any CLI tests.

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @bryanchriswhite 
**Co-owners**: @red-0ne

**Comment by bryanchriswhite** [comment]
Reopening as this calls for CLI test coverage, as currently written, which has not yet been implemented.

**Comment by bryanchriswhite** [comment]
Deprioritizing as claims & proofs are typically generated and broadcast by the relay miner, making CLI test coverage less urgent.

**Comment by Olshansk** [comment]
Closing this out since it's a lower priority relative to other deliverables.
